scripts/benchmark-write.cc

```diff
@@ -0,0 +1,72 @@
+#include <chrono>
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+#ifdef _WIN32
+#include <io.h>
+#define fsync _commit
+#else // linux / mac
+#include <unistd.h>
+#endif
+
+#include "gsd.h"
+
+using namespace std;
+
+int main(int argc, char** argv) // NOLINT
+{
+    const size_t n_keys = 65534;
+    const size_t n_frames = 1000;
+    const size_t key_size = 1;
+
+    vector<double> data(key_size);
+    vector<string> names;
+    for (size_t i = 0; i < n_keys; i++)
+    {
+        ostringstream s;
+        s << "log/hpmc/integrate/Sphere/quantity/" << i;
+        names.push_back(s.str());
+    }
+
+    cout << "Writing test.gsd with: " << n_keys << " keys, " << n_frames << " frames, "
+         << "and " << key_size << " double(s) per key" << endl;
+    gsd_handle handle;
+    gsd_create_and_open(&handle, "test.gsd", "app", "schema", 0, GSD_OPEN_APPEND, 0);
+    for (size_t frame = 0; frame < n_frames / 2; frame++)
+    {
+        for (auto const& name : names)
+        {
+            gsd_write_chunk(&handle, name.c_str(), GSD_TYPE_DOUBLE, key_size, 1, 0, &data[0]);
+        }
+        gsd_end_frame(&handle);
+    }
+    fsync(handle.fd);
+
+    auto t1 = chrono::high_resolution_clock::now();
```

Comment:
Why are you only timing half of the writes?

Reply:
To obtain timing for the typical case and exclude the time taken to write the initial namelist and index expansions.

Reply:
I figured something like that, but I was curious.